### ISSUE TITLE
Web support

### ISSUE THREAD

**Issue by skreborn** [mapbox/mapbox-maps-flutter#13]
The README states that "Web and desktop are not supported" - I'm wondering if that's a limitation as of this moment, or something that you don't intend to support even if the implementation is done by someone?

**Comment by yunikkk** [comment]
@skreborn hi, there is indeed no plan for Web/Desktop at the moment. We haven't considered supporting 3party implementation yet, however contributions are welcome and it could probably indeed happen when the repo is established and ready for it (part of the tooling for codegen etc is still private).

**Comment by skreborn** [comment]
Thank you for your response - I understand. I imagine the first step would be relying on a plugin interface and separate implementation packages instead of putting everything into a single package. I'll keep an eye on the project.

**Comment by cedvdb** [comment]
Wasn't there plans on the mapbox part to have one underlying code base for all platforms ? IE: wasm ?

**Comment by marianhlavac** [comment]
Having official Flutter plugin for Mapbox with web support would be great.

> it could probably indeed happen when the repo is established and ready for it (part of the tooling for codegen etc is still private)

@yunikkk Is there any way community can help?

**Comment by srmanc** [comment]
@marianhlavac @yunikkk @skreborn Community already did: https://github.com/flutter-mapbox-gl/maps

This is a fork of the original project from Mapbox. Community did web support, it works well. And this repo resembles the original structure very much - but it differs in one key aspect - the class implementations are all private, thus even if community wanted to port the existing web implementation, it can't.

I am sure the community (myself very much included) would be eager to prepare PRs for web, but Mapbox needs to enable us in doing so.

**Comment by fmt-Println-MKO** [comment]
the community version works well but has one problem: the web implementation is old, I received emails from Mapbox that the old API will be shut down. 
The latest release from the community version is 1 year ago. 
To have an officially supported version would be a lot more beneficial.
Also as Flutter is supporting more than just Android and iOS plugins should also support the other platforms as well.

**Comment by lukas-h** [comment]
any news here?

**Comment by aardrop** [comment]
Six months since any activity here, we would love an update as we're happy to contribute to this feature but can't move from the tobrun's open source package to this one until the web is supported.

**Comment by HHLiftBridge** [comment]
I am also interested in web support for this plugin. Has the stance on web changed?

**Comment by MelleLintje06** [comment]
Our project is unfortunately not compatable with the package mapbox_gl so were also very also interested in web support for this plugin.

**Comment by tfvtti** [comment]
We would like to move from mapbox_gl but our application has a web future so please add support for web.

**Comment by felix-ht** [comment]
@evil159 any updates on this? Are you willing to support the community with reviewing PRs and get them merged? I don't want to do any work on this repo to just have the PR rejected

**Comment by Fakerinc** [comment]
I would love this as well!

**Comment by aardrop** [comment]
> @evil159 any updates on this? Are you willing to support the community with reviewing PRs and get them merged? I don't want to do any work for this repo to just have the PR rejected/ignored

Exactly, I would also love to support this if it means we can incorporate it into this package.

**Comment by srmncnk** [comment]
In the meantime ESRI did this: https://pub.dev/packages/arcgis_map_sdk
They claim they support web, I haven't checked it out yet though.

**Comment by srmncnk** [comment]
Not to mention https://pub.dev/packages/maplibre_gl.

**Comment by JEThilenius** [comment]
> We would like to move from mapbox_gl but our application has a web future so please add support for web.

We are very interested in doing the same!

**Comment by aardrop** [comment]
Six-month check-in here: I'd love to contribute to getting the web plugged in here, but with @evil159 providing so much support, I don't want to build something that is out of schema or won't be merged. 

It's likely that the third-party open-source version won't support upcoming Flutter updates, and we can't move to this version unless it supports the web.

**Comment by evil159** [comment]
Hello everyone,

Apologies for the delayed response, and thank you so much for your offer to help implement web support for our Flutter SDK. We really appreciate your enthusiasm and willingness to contribute!

To clarify, we’re open to accepting contributions for web support, but we want to set the right expectations regarding maintenance. While we can introduce web support as an experimental feature, we’re not in a position to guarantee the same level of maintenance and long-term support that we provide for other platforms—at least not until we see higher demand for web usage.

This means web support would remain in an experimental state for the time being, with the possibility of full support in the future if the demand grows.

Since the project originated from our mobile team at Mapbox, the public API is naturally more aligned with mobile workflows. While many concepts are shared between mobile and web, there may be cases where the APIs don’t map perfectly.

To get the conversation started and explore the possibilities, would you be open to creating a proof-of-concept (PoC) pull request? This would give us a solid foundation for discussion and help shape the direction moving forward.

**Comment by srmncnk** [comment]
@evil159 Thank you for your response.

My two cents regarding pull requests - it may have changed since I last looked, but doing pull requests is probably impossible due to two reasons:
* some classes which needed to be overloaded are private
* you're using pigeon for wrapper code generation but did not disclose the template for it.

Otherwise I imagine we would be thrilled to have web even as an experimental feature. As long as we can contribute to it. It has been this way since the beginning of the flutter package after all.

**Comment by evil159** [comment]
@srmncnk To enable this we are willing to make the required accommodations in our SDK codebase, for the purposes of PoC I think it's fine to tamper with private APIs, expose private classes etc.
Yeah, currently the templates are in another repo, we'll have to move them here for this to work, I don't think it's required on PoC level, it doesn't have to be all-encompassing, just a web view with a map should be good enough, if there will be a way to set style uri/json that will be awesome.

**Comment by srmncnk** [comment]
@evil159 So you're saying you're willing to accept PRs that will modify private class names as well? I think we'd still appreciate access to the template though.

**Comment by aardrop** [comment]
@evil159 any confirmation on this^? The open source package is facing more and more deprecations, and we'll need to make a decision soon.

**Comment by Ronan-V** [comment]
I agree. How can we express our interest in web support so that you have an idea of how many we are?

**Comment by srmncnk** [comment]
We can't even do PRs for mobile packages because of undisclosed pigeon bindings, let alone web integration - which WOULD be possible since the underlying architecture hasn't changed much from tobrun's.

**Comment by pjleonard37** [comment]
We are excited to announce that we are actively working on adding web support to our Flutter SDK! 🎉 🎉 

This web support will be built on our [Mapbox GL JS library](https://docs.mapbox.com/mapbox-gl-js/guides/). Our initial goal is to release a basic Minimum Viable Product (MVP) in April, which will enable fundamental map functionality in a web browser. We look forward to your feedback on this initial release. Following the MVP, we plan to expand the supported features throughout the rest of the year.

**Comment by shivanshu877** [comment]
Sharing back from a production Flutter Web rollout.

We needed Mapbox in the browser while waiting for official web support, and
the cleanest path turned out to be conditional imports rather than a fork
of this plugin. The mobile path keeps using `mapbox_maps_flutter`
unchanged; the web path drops in a `dart:js_interop` wrapper around
[mapbox-gl-js v3](https://docs.mapbox.com/mapbox-gl-js/) that exposes the
same `MapboxMap` / `MapWidget` / `Style` / `Point` / `CameraOptions` /
`MapboxStyles` / `MapboxOptions` shapes this plugin already publishes — so
no call sites change.

Published at:
- pub.dev: <https://pub.dev/packages/mapbox_gl_flutter_web>
- source: <https://github.com/shivanshu877/mapbox_gl_flutter_web>

Coverage is the production subset (camera, GeoJSON sources with clustering,
Symbol + Line layers, runtime style image registration, queryRenderedFeatures,
style-import config for the Mapbox Standard `lightPreset`, scale-bar /
compass / gesture toggles). Not covered: annotations API, location puck,
terrain / fog, raster sources, 3D model layers.

Two non-obvious gotchas worth documenting somewhere if web support ever
lands in this plugin:

1. **Public token required.** Browsers can't use secret (`sk.…`) tokens —
   the tile API silently rejects them and you get a black canvas with no
   error in the console. Worth a `setAccessToken` validation guard.

2. **Container layout timing.** `new mapboxgl.Map({container})` locks the
   canvas at the 400×300 fallback if the container has zero layout at
   construction time. Flutter Web's `HtmlElementView` mounts the div before
   the layout pass runs, so the naive impl always falls into that trap. Two
   frame delays plus a `ResizeObserver` calling `map.resize()` fixed it for
   us.

Happy to fold this into an official platform-implementation if there's
interest, but understand if web isn't on the roadmap — putting it on pub.dev
as a community plugin keeps the option open for anyone who needs it now.

**Comment by sethjhs9** [comment]
Since this feature is so delayed, I’m taking a look at maplibre. Looks very promising!
https://pub.dev/packages/maplibre

**Comment by oosafff** [comment]
**Web support for the Mapbox Maps SDK for Flutter is now in public preview!** 

Thank you all for your patience on this one. Web support has been the most requested feature on this repo, so thank you to everyone who filed and upvoted issues.

**Packages (3.0.0-alpha.1)**
- mapbox_maps_flutter (app-facing) — https://pub.dev/packages/mapbox_maps_flutter/versions/3.0.0-alpha.1
- mapbox_maps_flutter_mobile — https://pub.dev/packages/mapbox_maps_flutter_mobile/versions/3.0.0-alpha.1
- mapbox_maps_flutter_web — https://pub.dev/packages/mapbox_maps_flutter_web/versions/3.0.0-alpha.1
- mapbox_maps_flutter_platform_interface — https://pub.dev/packages/mapbox_maps_flutter_platform_interface/versions/3.0.0-alpha.1

**Installation guide:** https://docs.mapbox.com/flutter/maps/guides/install-v3/

**What works on web today:** map display and styling, camera and animations, style layers and sources, gestures, the location component, and viewport. Typed interactions support tap for now. Annotations, ornaments, feature query, and snapshot are still landing (use style layers for markers in the meantime). Offline and native debug features do not apply on web.

This is an alpha release, so it's not yet feature complete. The API can still change, and we are filling the web gaps as we move toward general availability later this year. Please try it and tell us what features you want us to prioritize.

Found a bug or hitting a gap? Open an issue on this repo.
Question about your account or access token? Contact [Mapbox support](https://support.mapbox.com/).

**Comment by eugsokolov** [comment]
@oosafff thanks for sharing that is great news! much excited!

Can you share plans regarding wasm support? is that on the roadmap?

**Comment by srmncnk** [comment]
Can you share the benefits of this against vibecoding a target ed pure dart2js solution involving existing mapbox/maplibre js bindings?
